packages/grafana-data/src/dataframe/processDataFrame.ts

```diff
@@ -26,12 +27,12 @@ import { getFieldDisplayName } from '../field/fieldState';
 
 function convertTableToDataFrame(table: TableData): DataFrame {
   const fields = table.columns.map(c => {
-    const { text, ...disp } = c;
+    const { text, type, ...disp } = c as MutableColumn;
     return {
       name: text, // rename 'text' to the 'name' field
       config: (disp || {}) as FieldConfig,
       values: new ArrayVector(),
-      type: FieldType.other,
+      type: type && Object.values(FieldType).includes(type as FieldType) ? (type as FieldType) : FieldType.other,
```

Comment:
anre there cases where the `.type` is not a FieldType?

Reply:
Not that I know of, but if there are, the behavior should be the same as before this change